components/gitpod-db/migrate_gcp.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+# Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+# Licensed under the GNU Affero General Public License (AGPL).
+# See License-AGPL.txt in the project root for license information.
+
+# This scipt connects via Google's cloud_sql_proxy to a database and runs the db-migrations
+
+# ENV variables for configuration:
+# * GOOGLE_APPLICATION_CREDENTIALS_DATA: contents of the crendetials files that cloud_sql_proxy uses for authentication
+# * GCP_DATABASE: database name
+# * DB_PASSWORD: database password
+
+# Example usage:
+# docker run --rm \
+#            --env GOOGLE_APPLICATION_CREDENTIALS_DATA='...' \
+#            --env GCP_DATABASE="gitpod-foobar:europe-west1:gitpod-foobar-baz" \
+#            --env DB_PASSWORD="..." \
+#            gcr.io/gitpod-core-dev/build/db-migrations:x1 /app/migrate_gcp.sh
+
+set -euo pipefail
+
+echo "$GOOGLE_APPLICATION_CREDENTIALS_DATA" > /tmp/gcp.json
```

Comment:
some check for the existence of those env vars together with a sensible usage message would be great.

Reply:
[`set -u`](https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html) automates both for us:
> Treat unset variables and parameters other than the special parameters ‘@’ or ‘*’ as an error when performing parameter expansion. An error message will be written to the standard error, and a non-interactive shell will exit.

That's enough IMHO unless we want error messages that go beyond saying "please set this env var".